spec/models/async_transaction/vet360/base_spec.rb

```diff
@@ -82,11 +82,55 @@
   end
 
   describe '.fetch_transaction' do
-    let(:service) { ::Vet360::ContactInformation::Service.new(user) }
     it 'raises an error if passed unrecognized transaction' do
       expect do
         AsyncTransaction::Vet360::Base.fetch_transaction(Struct.new('Surprise'), nil)
       end.to raise_exception(RuntimeError)
     end
   end
+
+  describe '.refresh_transaction_statuses()' do
+    let(:user) { build(:user, :loa3) }
+    let(:transaction1) do
+      create(:address_transaction,
+             transaction_id: '0faf342f-5966-4d3f-8b10-5e9f911d07d2',
+             user_uuid: user.uuid,
+             status: AsyncTransaction::Vet360::Base::COMPLETED)
+    end
+    let(:service) { ::Vet360::ContactInformation::Service.new(user) }
+
+    before do
+      # vet360_id appears in the API request URI so we need it to match the cassette
+      allow_any_instance_of(Mvi).to receive(:response_from_redis_or_service).and_return(
+        MVI::Responses::FindProfileResponse.new(
+          status: MVI::Responses::FindProfileResponse::RESPONSE_STATUS[:ok],
+          profile: build(:mvi_profile, vet360_id: '1')
+        )
+      )
+    end
+
+    it 'does not return completed transactions (whose status has not changed)' do
+      transactions = AsyncTransaction::Vet360::Base.refresh_transaction_statuses(user, service)
+      expect(transactions).to eq([])
+    end
+
+    it 'returns only the most recent transaction address/telephone/email transaction' do
+      create(:email_transaction,
+             transaction_id: 'foo',
+             user_uuid: user.uuid,
+             transaction_status: 'RECEIVED',
+             status: AsyncTransaction::Vet360::Base::REQUESTED,
+             created_at: Time.zone.now - 1)
+      create(:email_transaction,
```

Comment:
Small nit.  If you set this transaction to a variable, you can call it and ref it's `transaction_id` attr in the expectation, dynamically.  Perhaps that saves the magic number comparison?

For example:

```ruby
recent_transaction = create(:email_transaction, ....)

expect(transactions.first.transaction_id).to eq recent_transaction.transaction_id
```